File: Top Interview 150/Sliding Window/3_Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        map<char, int> seen;
        int count = 0;
        int l = 0;
        char curr_char;
        for (int i=0; i<s.length(); i++) {
            curr_char = s[i];
            if (seen.count(curr_char) && seen[curr_char] >= l) {
                l = seen[curr_char] + 1;
            }
            else {
                count = max(count, i-l+1);
            }
            seen[curr_char] = i;
        }
        return count;
    }
};

File: Top Interview 150/Sliding Window/3_Longest Substring Without Repeating Characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        seen = {}
        count = 0
        l = 0
        for i in range(len(s)):
            curr_char = s[i]
            if curr_char in seen and seen[curr_char] >= l:
                l = seen[curr_char] + 1
            else:
                count = max(count, i-l+1)
            seen[curr_char] = i
        return count